prov/sockets/include/sock.h

```diff
@@ -196,6 +196,7 @@ struct sock_fabric {
 struct sock_conn {
         int sock_fd;
         int connected;
+        int released;
```

Comment:
instead of adding "released", can we indicate this in the sock_fd itself (say, -1) ?